src/lib.rs

```diff
@@ -2467,6 +2450,54 @@ impl Build {
             println!("{}", s);
         }
     }
+
+    #[cfg(target_os = "macos")]
+    fn fix_env_for_apple_os(&self, cmd: &mut Command) -> Result<(), Error> {
+        let target = self.get_target()?;
+        let host = self.get_host()?;
+        if host.contains("apple-darwin") && target.contains("apple-darwin") {
+            // Replace the `SDKROOT` environment variable if it's clearly set for the wrong platform, which
+            // may occur when we're linking a custom build script while targeting iOS for example.
+            // Removing is not enough, because of cc from XCode can not find include files in such case
+            if let Ok(sdkroot) = env::var("SDKROOT") {
+                if sdkroot.contains("iPhone") {
```

Comment:
Should this perhaps be inverted? It seems like Apple is inevitably going to release more sdks, so it might be good to not have to update this every time.

Reply:
Done